runtime/wtp.c

```diff
@@ -437,6 +445,11 @@ wtpStartWrkr(wtp_t *pThis)
 		  wtpGetDbgHdr(pThis), iState,
 		  ATOMIC_FETCH_32BIT(&pThis->iCurNumWrkThrd, &pThis->mutCurNumWrkThrd));
 
+        /* wait for the new thread to initialize its signal mask and
+         * cancelation cleanup handler before proceeding
+         */
+        d_pthread_cond_wait(&pThis->condThrdInitDone, &pThis->mutWtp);
```

Comment:
So I realize that I am a bit late to the game here, but the condition wait call does not appear to be in a loop.  Perhaps I am missing the loop?  POSIX allows for spurious wake ups to occur for callers of `pbench_cond_wait()`, so the recommended sequence/hand shake is to:

```
WAITER:
lock
   loop
      check predicate (memory location value)
      if not expected state of predicate
        then call condition wait (which unlocks mutex and reacquires on wakeup)
        else exit loop
unlock

SIGNALER:
lock
    change wait predicate
    signal or broadcast
unlock
```

This code does not appear to do that, so it is likely wide open to race conditions.